### DISCUSSION TITLE
Auto ID

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#7105] [Help] [answered]
Hey, 

we have a serious problem with my Auto ID column.
Creating a new row duplicated 2 Auto ID numbers. 
Meaning: 
- We created 2 contracts with Auto ID numbers 56 & 57. 
- 2 new created rows then had the same numbers. 

![image](https://user-images.githubusercontent.com/95284278/183023894-331b260c-c9af-4420-afad-b8fbee57a7e9.png)

Thanks!

**Comment by melohagan** [comment]
Hey @kasgithub8 

Did you manually set the Auto IDs at any point? I.e. when you created the two contracts with Auto ID numbers 56 & 57, did you actually specify these numbers, or were they automatically generated?

**Reply by kasgithub8** [reply]
Hi @melohagan,

I didn't.
My guess is that if 1 user creates a "contract" and another user also creates one, without the form being refreshed, it uses the same ID.

**Reply by melohagan** [reply]
Nice hypothesis, I can look into that. Just to double check, you are using the internal Budibase DB?

**Reply by kasgithub8** [reply]
@melohagan Thanks and yes, I am.

**Reply by melohagan** [reply]
I investigated this:
> My guess is that if 1 user creates a "contract" and another user also creates one, without the form being refreshed, it uses the same ID.

But I wasn't getting any issue for two different users. To confirm again, you are not assigning anything to the Auto ID column, such as:
![Screenshot 2022-08-16 at 14 13 40](https://user-images.githubusercontent.com/101575380/184888324-8c2186cf-45f0-40b0-b973-e939dc5f4172.png)
It definitely is being left to being automatically assigned?

When you created two new contracts, was that two individual form saves, or does the one form save two rows?

**Reply by kasgithub8** [reply]
> But I wasn't getting any issue for two different users. To confirm again, you are not assigning anything to the Auto ID column, such as: ![Screenshot 2022-08-16 at 14 13 40](https://user-images.githubusercontent.com/101575380/184888324-8c2186cf-45f0-40b0-b973-e939dc5f4172.png) It definitely is being left to being automatically assigned?

They're being assigned automatically.
> When you created two new contracts, was that two individual form saves, or does the one form save two rows?

I didn't create them myself, but the users I created this app for are both using the same form.
What I have noticed in a test app is when you f.e. have the app data and the published app open it's two different sessions.
So when you create a contract it won't pop up in the table. You have to reload.

**Comment by melohagan** [comment, marked as answer]
@kasgithub8 Perfect thank you, I was able to reproduce the issue. You're absolutely right in that it is when one user is on dev, and another is on the published app. 

Behind the scenes Budibase is doing a sync between the production and dev databases, which must be causing the issue with the auto ids. 

I have raised an issue here: https://github.com/Budibase/budibase/issues/7311

**Comment by DWilly92** [comment]
Issue #7311 was closed over 6 months ago, indicating that an update was forthcoming that would resolve this issue.

I am seeing this issue in my environment, running version 2.15.7. I have a form set up with an Auto ID column, no automations, and my users do not have access to the dev site, but IDs are being duplicated on forms my users are submitting. I have just given my users access to the form, and now this issue has cropped up when I was reviewing their data input over this past week. This will be a major issue, if we can't rely on the Auto ID column.

@melohagan @Rory-Powell - Can we have an update on this?

![image](https://github.com/Budibase/budibase/assets/26339291/b82570c7-b60c-4cb2-87a2-203d7e8a028d)

**Reply by melohagan** [reply]
Hey @DWilly92 Can you show me a screenshot of the **Save Row** action of your form(s) please?

**Reply by DWilly92** [reply]
Certainly! 😊 - This is a 7 step form with a basic save action. The form schema is set to the smr table as well as the save action, and the form type is "Create".

![image](https://github.com/Budibase/budibase/assets/26339291/f87896b6-5039-41e3-a8ad-38cfcf56dc5f)

**Reply by melohagan** [reply]
Hmm that looks fine 🤔 

Is this something you have noticed occurring recently? Or has this been ongoing for a while? Is it a frequent occurrence?

**Reply by DWilly92** [reply]
This is something I just noticed when reviewing the data that some of our users inputted over this past week. It'll be a few days before they input more, and I'll have a better picture of what's going on at that point, but I wanted to get ahead of the curve.

I would just submit some forms myself to test but to my knowledge there's no way to reset the ID numbers in that AutoID column, and I'd like to keep them consistent if possible.

**Reply by melohagan** [reply]
Are you a cloud user, or self-hosted?

**Reply by DWilly92** [reply]
Self-hosted, running version 2.15.7
